compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -403,7 +404,7 @@ class TypeErasure(isJava: Boolean, semiEraseVCs: Boolean, isConstructor: Boolean
             case tr :: trs1 =>
               assert(!tr.classSymbol.is(Trait), cls)
               val tr1 = if (cls is Trait) defn.ObjectType else tr
-              tr1 :: trs1.filterNot(_ isRef defn.ObjectClass)
+              tr1 :: trs1.filterNot(x => x.isRef(defn.ObjectClass) || x.isRef(defn.PhantomClass))
```

Comment:
This clearly need a comment. It is erasing the phantom lattice (i.e. removing the `scala.Phantom` trait from the object defining the lattice.